# Importing libraries and data

In [42]:
import pandas as pd

demo = pd.read_sas("../Data/Raw/DEMO_L.xpt")
dpq = pd.read_sas("../Data/Raw/DPQ_L.xpt")

# Initial observations

## Current goal
Current goal is to become familiar with the dataset and identify which variables are relevant for investigating the factors associated with depressive symptoms 

## Current questions
- What variables do we have?
- Is there any missing data?
- What data limitations do we have?
- How many individuals from demographic data completed the depression screener?

In [2]:
demo.head()

,SEQN,SDDSRVYR,RIDSTATR,RIAGENDR,RIDAGEYR,RIDAGEMN,RIDRETH1,RIDRETH3,RIDEXMON,RIDEXAGM,...,DMDHRGND,DMDHRAGZ,DMDHREDZ,DMDHRMAZ,DMDHSEDZ,WTINT2YR,WTMEC2YR,SDMVSTRA,SDMVPSU,INDFMPIR
0,130378.0,12.0,2.0,1.0,43.0,NaN,5.0,6.0,2.0,NaN,...,NaN,NaN,NaN,NaN,NaN,50055.450807,54374.463898,173.0,2.0,5.00
1,130379.0,12.0,2.0,1.0,66.0,NaN,3.0,3.0,2.0,NaN,...,NaN,NaN,NaN,NaN,NaN,29087.450605,34084.721548,173.0,2.0,5.00
2,130380.0,12.0,2.0,2.0,44.0,NaN,2.0,2.0,1.0,NaN,...,NaN,NaN,NaN,NaN,NaN,80062.674301,81196.277992,174.0,1.0,1.41
3,130381.0,12.0,2.0,2.0,5.0,NaN,5.0,7.0,1.0,71.0,...,2.0,2.0,2.0,3.0,NaN,38807.268902,55698.607106,182.0,2.0,1.53
4,130382.0,12.0,2.0,1.0,2.0,NaN,3.0,3.0,2.0,34.0,...,2.0,2.0,3.0,1.0,2.0,30607.519774,36434.146346,182.0,2.0,3.60


In [3]:
dpq.head()

,SEQN,DPQ010,DPQ020,DPQ030,DPQ040,DPQ050,DPQ060,DPQ070,DPQ080,DPQ090,DPQ100
0,130378.0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
1,130379.0,5.397605e-79,5.397605e-79,1.000000e+00,5.397605e-79,5.397605e-79,5.397605e-79,5.397605e-79,5.397605e-79,5.397605e-79,5.397605e-79
2,130380.0,5.397605e-79,5.397605e-79,1.000000e+00,1.000000e+00,5.397605e-79,5.397605e-79,5.397605e-79,5.397605e-79,5.397605e-79,5.397605e-79
3,130386.0,5.397605e-79,5.397605e-79,5.397605e-79,5.397605e-79,5.397605e-79,1.000000e+00,5.397605e-79,5.397605e-79,5.397605e-79,5.397605e-79
4,130387.0,5.397605e-79,5.397605e-79,5.397605e-79,5.397605e-79,5.397605e-79,5.397605e-79,5.397605e-79,5.397605e-79,5.397605e-79,NaN


In [5]:
demo.shape, dpq.shape

((11933, 27), (6337, 11))

In [4]:
demo.info()

<class 'pandas.DataFrame'>
RangeIndex: 11933 entries, 0 to 11932
Data columns (total 27 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   SEQN      11933 non-null  float64
 1   SDDSRVYR  11933 non-null  float64
 2   RIDSTATR  11933 non-null  float64
 3   RIAGENDR  11933 non-null  float64
 4   RIDAGEYR  11933 non-null  float64
 5   RIDAGEMN  377 non-null    float64
 6   RIDRETH1  11933 non-null  float64
 7   RIDRETH3  11933 non-null  float64
 8   RIDEXMON  8860 non-null   float64
 9   RIDEXAGM  2787 non-null   float64
 10  DMQMILIZ  8301 non-null   float64
 11  DMDBORN4  11914 non-null  float64
 12  DMDYRUSR  1875 non-null   float64
 13  DMDEDUC2  7794 non-null   float64
 14  DMDMARTZ  7792 non-null   float64
 15  RIDEXPRG  1503 non-null   float64
 16  DMDHHSIZ  11933 non-null  float64
 17  DMDHRGND  4115 non-null   float64
 18  DMDHRAGZ  4124 non-null   float64
 19  DMDHREDZ  3746 non-null   float64
 20  DMDHRMAZ  4020 non-null   float64
 21  

In [5]:
demo.describe()

,SEQN,SDDSRVYR,RIDSTATR,RIAGENDR,RIDAGEYR,RIDAGEMN,RIDRETH1,RIDRETH3,RIDEXMON,RIDEXAGM,...,DMDHRGND,DMDHRAGZ,DMDHREDZ,DMDHRMAZ,DMDHSEDZ,WTINT2YR,WTMEC2YR,SDMVSTRA,SDMVPSU,INDFMPIR
count,11933.000000,11933.0,11933.000000,11933.000000,1.193300e+04,3.770000e+02,11933.000000,11933.000000,8860.000000,2.787000e+03,...,4115.000000,4124.000000,3746.000000,4020.000000,2127.000000,11933.000000,1.193300e+04,11933.000000,11933.000000,9.892000e+03
mean,136344.000000,12.0,1.742479,1.532808,3.831786e+01,1.162865e+01,3.104584,3.320540,1.520203,1.219085e+02,...,1.564277,2.539525,2.171383,1.381095,2.275035,27404.135674,2.740414e+04,179.918294,1.491746,2.708174e+00
std,3444.904716,0.0,0.437287,0.498943,2.560199e+01,6.805429e+00,1.076346,1.518379,0.499620,6.715865e+01,...,0.495912,0.641094,0.660132,0.681421,0.694050,19449.155209,2.796296e+04,4.309624,0.499953,1.670119e+00
min,130378.000000,12.0,1.000000,1.000000,5.397605e-79,5.397605e-79,1.000000,1.000000,1.000000,5.397605e-79,...,1.000000,1.000000,1.000000,1.000000,1.000000,4584.463196,5.397605e-79,173.000000,1.000000,5.397605e-79
25%,133361.000000,12.0,1.000000,1.000000,1.300000e+01,6.000000e+00,3.000000,3.000000,1.000000,6.600000e+01,...,1.000000,2.000000,2.000000,1.000000,2.000000,14331.753594,5.397605e-79,176.000000,1.000000,1.180000e+00
50%,136344.000000,12.0,2.000000,2.000000,3.700000e+01,1.100000e+01,3.000000,3.000000,2.000000,1.220000e+02,...,2.000000,2.000000,2.000000,1.000000,2.000000,21670.185970,2.171785e+04,180.000000,1.000000,2.500000e+00
75%,139327.000000,12.0,2.000000,2.000000,6.200000e+01,1.700000e+01,4.000000,4.000000,2.000000,1.795000e+02,...,2.000000,3.000000,3.000000,2.000000,3.000000,33831.331556,3.834115e+04,184.000000,2.000000,4.500000e+00
max,142310.000000,12.0,2.000000,2.000000,8.000000e+01,2.400000e+01,5.000000,7.000000,2.000000,2.390000e+02,...,2.000000,4.000000,3.000000,3.000000,3.000000,170968.343177,2.271083e+05,187.000000,2.000000,5.000000e+00


In [6]:
dpq.info()

<class 'pandas.DataFrame'>
RangeIndex: 6337 entries, 0 to 6336
Data columns (total 11 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   SEQN    6337 non-null   float64
 1   DPQ010  5519 non-null   float64
 2   DPQ020  5518 non-null   float64
 3   DPQ030  5516 non-null   float64
 4   DPQ040  5514 non-null   float64
 5   DPQ050  5513 non-null   float64
 6   DPQ060  5510 non-null   float64
 7   DPQ070  5508 non-null   float64
 8   DPQ080  5508 non-null   float64
 9   DPQ090  5506 non-null   float64
 10  DPQ100  4167 non-null   float64
dtypes: float64(11)
memory usage: 544.7 KB


In [7]:
dpq.describe()

,SEQN,DPQ010,DPQ020,DPQ030,DPQ040,DPQ050,DPQ060,DPQ070,DPQ080,DPQ090,DPQ100
count,6337.000000,5.519000e+03,5.518000e+03,5.516000e+03,5.514000e+03,5.513000e+03,5.510000e+03,5.508000e+03,5.508000e+03,5.506000e+03,4.167000e+03
mean,136348.738362,5.109621e-01,4.786154e-01,7.935098e-01,8.601741e-01,5.042627e-01,4.110708e-01,3.948802e-01,2.165940e-01,8.209226e-02,4.792417e-01
std,3439.132476,9.466645e-01,8.387239e-01,1.003693e+00,9.651269e-01,8.605773e-01,7.869793e-01,7.675240e-01,6.850770e-01,4.381644e-01,7.401059e-01
min,130378.000000,5.397605e-79,5.397605e-79,5.397605e-79,5.397605e-79,5.397605e-79,5.397605e-79,5.397605e-79,5.397605e-79,5.397605e-79,5.397605e-79
25%,133337.000000,5.397605e-79,5.397605e-79,5.397605e-79,5.397605e-79,5.397605e-79,5.397605e-79,5.397605e-79,5.397605e-79,5.397605e-79,5.397605e-79
50%,136397.000000,5.397605e-79,5.397605e-79,5.397605e-79,1.000000e+00,5.397605e-79,5.397605e-79,5.397605e-79,5.397605e-79,5.397605e-79,5.397605e-79
75%,139310.000000,1.000000e+00,1.000000e+00,1.000000e+00,1.000000e+00,1.000000e+00,1.000000e+00,1.000000e+00,5.397605e-79,5.397605e-79,1.000000e+00
max,142310.000000,9.000000e+00,9.000000e+00,9.000000e+00,9.000000e+00,9.000000e+00,9.000000e+00,9.000000e+00,9.000000e+00,9.000000e+00,9.000000e+00


### Demographic (DEMO) dataset overview

The 'DEMO' dataset provides the demographic information, such as age, sex, race, education, poverty status, family status and other background characteristics. 

There are 27 variables. Let's briefly review each of them.
1. SEQN - Respondent sequence number, we can use SEQN to connect with other tables.
2. SDDSRVYR - Data release cycle. A value of '12' means August 2021-August 2023 data. Our whole dataset is based on this cycle.
3. RIDSTATR - Interview/Examination status. 
   - Status '1' identifies a participant was only interviewed but never participated in the examination.
   - Status '2' identifies a participant was both interviewed and examined in the mobile examination center (MEC).
4. RIAGENDR - Gender.
   - '1' is a male.
   - '2' is a female.
5. RIDAGEYR - Age in years at the time of screening. Individuals 80 and over are top-coded at 80 years of age.
6. RIDAGEMN - Age in months at the time of screening. Field used only for children aged 24 months or younger at the time of exam (or screening if not examined).
7. RIDRETH1 (old parameter, to link with previous studies) - Race-ethnicity categories 
   - '1' - '4' - variables are the same as `RIDRETH3`
   - '5' - used only in RIDRETH1 for Asian participants and  other non-Hispanic races.
8. RIDRETH3 - Race-ethnicity categories
   -  '1' - respondents who self-identified as "Mexican American".
   -  '2' - participants who self-identified as "Hispanic".
   -  '3' - individuals who self-reported their race as "non-Hispanic White".
   -  '4' - those who self-reported their race as "non-Hispanic Black".
   -  '6' - those who self-reported their race as "non-Hispanic Asian".
   -  '7' - other non-Hispanic races, including non-Hispanic multiracial.
9. RIDEXMON - Examination period. Six month time period when the examination was performed.
   - '1' - November 1 through April 30.
   - '2' - May 1 through October 31.
10. RIDEXAGM - Age in months at the time of examination for people aged 19 years or younger.
11. DMQMILIZ - Whether the respondent has ever served active duty in US Armed Forces. 
   - '1' - Yes
   - '2' - No
   - '7' - Refused (to answer)
   - '9' - Don't know
12. DMDBORN4 -  Country of birth. 
   - '1' - Born in 50 U.S. states or Washington, DC.
   - '2' - Born in other countries, including U.S. territories
   - '77' - Refused (to answer)
   - '99' - Don't know
13. DMDYRUSR - Length of time living in the US.
   - '1' - Less than 1 year
   - '2' - 1 to 4 years
   - '3' - 5 to 9 years
   - '4' - 10 to 14 years
   - '5' - 15 to 19 years
   - '6' - 20 years or more
   - '77' - Refused (to answer)
   - '99' - Don't know
14. DMDEDUC2 - The highest level of education the respondent has achieved. Collected for people above 20.
   - '1' - Less than 9th grade
   - '2' - 9-11th grade (Includes 12th grade with no diploma)
   - '3' - High school graduate/GED or equivalent
   - '4' - Some college or AA degree
   - '5' - College graduate or above
   - '7' - Refused (to answer)
   - '9' - Don't know
15. DMDMARTZ - Marital status. Collected for people above 20.
   - '1' - Married/Living with partner
   - '2' - Widowed/Divorced/Separated
   - '3' - Never married
   - '77' - Refused (to answer)
   - '99' - Don't know
16. RIDEXPRG - Pregnancy status at exam for females only 20 - 44 years.
   - '1' - Yes, positive lab pregnancy test or self-reported pregnant at exam
   - '2' - The participant was not pregnant at exam
   - '3' - Cannot ascertain if the participant is pregnant at exam
17. DMDHHSIZ -  Total number of people in the household.
   - '1' - '6' - range from 1 to 6 people in the household
   - '7' - 7 or more people in the household

   The next variables with the `DMDHR` prefix contain the background information of the household reference person (parent or legal guardian) for youth participants aged 0–19:
   
18. DMDHRGND - Household Reference Person’s gender.
   - '1' - Male.
   - '2' - Female.
19. DMDHRAGZ - Household Reference Person’s age in years.
   - '1' - <20 years
   - '2' - 20-39 years
   - '3' - 40-59 years
   - '4' - 60+ years
20. DMDHREDZ - Household Reference Person’s education level.
   - '1' - Less than high school degree
   - '2' - High school grad/GED or some college/AA degree
   - '3' - College graduate or above
21. DMDHRMAZ - Household Reference Person’s marital status.
   - '1' - Married/Living with partner
   - '2' - Widowed/Divorced/Separated
   - '3' - Never Married
22. DMDHSEDZ - Household Reference Person’s spouse’s education level
   - '1' - Less than high school degree
   - '2' - High school grad/GED or some college/AA degree
   - '3' - College graduate or above
23. WTINT2YR - Full sample 2-year interview weight. 
24. WTMEC2YR - Full sample 2-year MEC exam weight. The value is 0 in case the individual has never been examined.

*Note: The 2-year sample weights (`WTINT2YR`, `WTMEC2YR`, depending on the analysis) should be used for all NHANES August 2021-August 2023 analyses to estimate characteristics of the entire US population. **However, this exploratory analysis uses unweighted data. Therefore, the results should be interpreted as describing the NHANES study sample rather than the entire US population.***

25. SDMVSTRA - Masked variance pseudo-stratum.
26. SDMVPSU - Masked variance pseudo-PSU.

*Note: Masked Variance Units (`SDMVSTRA`, `SDMVPSU`) are variables describing the complex NHANES sampling design. They are required for survey-weighted statistical analysis **but are not used in the current exploratory analysis.***

27. INDFMPIR - Ratio of family income to the federal poverty threshold. Values below 1 indicate that the family's income is below the poverty line.

### Selecting variables for the analysis

The DEMO dataset contains 27 variables, described above. However, not all of these variables are relevant to the current research question. In this section, we identify the variables that will be used in the exploratory analysis and explain why they were selected. 

| Variable | Use? | Reason |
| :--- | :---: | :--- |
| SEQN | Yes | Merge datasets |
| SDDSRVYR | No | All datasets in current project have the same cycle |
| RIDSTATR | Yes | Required for analyses involving examination data (future questions) |
| RIAGENDR | Yes | Demographic characteristic of interest (sex) |
| RIDAGEYR | Yes | Demographic characteristic of interest (age) |
| RIDAGEMN | No | Age for children, we'll focus on adults above 20 |
| RIDRETH1 | No | Old variable to link with previous cycles |
| RIDRETH3 | Yes | Demographic characteristic of interest (race/ethnicity) |
| RIDEXMON | Yes | Examination period, may be useful for examination variables (future questions) |
| RIDEXAGM | No | Age for children in the examination period, we'll focus on adults above 20 |
| DMQMILIZ | Yes | May be associated with mental health outcomes (served in the army) |
| DMDBORN4 | Yes | Demographic characteristic of interest (place of birth) |
| DMDYRUSR | Yes | May be associated with adaptation and mental health (length of time living in the US) |
| DMDEDUC2 | Yes | Indicator of socioeconomic status (education) |
| DMDMARTZ | Yes | May be associated with social support and mental health (marital) |
| RIDEXPRG | Yes | Pregnancy may influence depressive symptoms (future questions)|
| DMDHHSIZ | Yes | Household composition may be associated with mental health |
| DMDHRGND | No | Relevant only for people under 20 |
| DMDHRAGZ | No | Relevant only for people under 20 |
| DMDHREDZ | No | Relevant only for people under 20 |
| DMDHRMAZ | No | Relevant only for people under 20 |
| DMDHSEDZ | No | Relevant only for people under 20 |
| WTINT2YR | No | Interview weight, documented but not used in the analysis |
| WTMEC2YR | No | Exam weight, documented but not used in the analysis |
| SDMVSTRA | No | Required only for survey-weighted inference |
| SDMVPSU | No | Required only for survey-weighted inference |
| INDFMPIR | Yes | Indicator of socioeconomic status (poverty) |

In [115]:
demo_adults = demo[['SEQN', 'RIDSTATR', 'RIAGENDR', 'RIDAGEYR', 'RIDRETH3', 'RIDEXMON', 'DMQMILIZ', 'DMDBORN4',
       'DMDYRUSR', 'DMDEDUC2', 'DMDMARTZ', 'RIDEXPRG','DMDHHSIZ', 'INDFMPIR']].copy()

In [116]:
demo_adults = demo_adults[demo_adults['RIDAGEYR']>=20]
demo_adults.info()

<class 'pandas.DataFrame'>
Index: 7809 entries, 0 to 11932
Data columns (total 14 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   SEQN      7809 non-null   float64
 1   RIDSTATR  7809 non-null   float64
 2   RIAGENDR  7809 non-null   float64
 3   RIDAGEYR  7809 non-null   float64
 4   RIDRETH3  7809 non-null   float64
 5   RIDEXMON  6064 non-null   float64
 6   DMQMILIZ  7794 non-null   float64
 7   DMDBORN4  7793 non-null   float64
 8   DMDYRUSR  1574 non-null   float64
 9   DMDEDUC2  7794 non-null   float64
 10  DMDMARTZ  7792 non-null   float64
 11  RIDEXPRG  1503 non-null   float64
 12  DMDHHSIZ  7809 non-null   float64
 13  INDFMPIR  6489 non-null   float64
dtypes: float64(14)
memory usage: 915.1 KB


### DEMO data selection summary
The original DEMO dataset contained 11933 participants. After selecting adults aged 20 years or older, the resulting dataset contained 7809 participants (65.4% out of original sample).

In [117]:
print(f'Original sample: {len(demo)}')
print(f'Analytical sample: {len(demo_adults)}')
print(f'Excluded: {len(demo) - len(demo_adults)}')
print(f'Percentage of original sample: {len(demo_adults)/len(demo):.2%}')

Original sample: 11933
Analytical sample: 7809
Excluded: 4124
Percentage of original sample: 65.44%


### Mental health - depression screener (DPQ) dataset overview

The 'DPQ' dataset contains the responses the Patient Health Questionnaire (also, PHQ-9), which provides the frequency of depression symptoms over the past two weeks. There are 9 questions for each depression symptom, the participants scored how frequently they had experienced those symptoms recently from 0 to 3. In addition, there was 1 question (`DPQ100`) about how much those symptoms bothered the participant.

A total score, ranging from 0 to 27, can be calculated for participants who have complete answers to the symptom questions (`DPQ010` - `DPQ090`). Major depression and depression severity can be assessed using pre-defined cut-points, described in the references.

There are 11 variables. Let's briefly review each of them.

1. SEQN - Respondent sequence number, we can use SEQN to connect with other tables.
2. DPQ010 - Have a little interest in doing things.
    
    _The question is "Over the last 2 weeks, how often have you been bothered by the following problems: little interest or pleasure in doing things? Would you say..."_

3. DPQ020  - Feeling down, depressed, or hopeless.

    _The question is "Over the last 2 weeks, how often have you been bothered by the following problems: feeling down, depressed, or hopeless?"_

4. DPQ030  - Trouble sleeping or sleeping too much.

    _The question is "Over the last 2 weeks, how often have you been bothered by the following problems: trouble falling or staying asleep, or sleeping too much?"_

5. DPQ040  - Feeling tired or having little energy.

    _The question is "Over the last 2 weeks, how often have you been bothered by the following problems: feeling tired or having little energy?"_

6. DPQ050 - Poor appetite or overeating.
    
    _The question is "Over the last 2 weeks, how often have you been bothered by the following problems: poor appetite or overeating?"_

7. DPQ060 - Feeling bad about yourself.

    _The question is "Over the last 2 weeks, how often have you been bothered by the following problems: feeling bad about yourself - or that you are a failure or have let yourself or your family down?"_

8. DPQ070 - Trouble concentrating on things.

    _The question is "Over the last 2 weeks, how often have you been bothered by the following problems: trouble concentrating on things, such as reading the newspaper or watching TV?"_

9. DPQ080 - Moving or speaking slowly or too fast.

    _The question is "Over the last 2 weeks, how often have you been bothered by the following problems: moving or speaking so slowly that other people could have noticed? Or the opposite - being so fidgety or restless that you have been moving around a lot more than usual?"_

10. DPQ090 - Thought you would be better off dead.

    _The question is "Over the last 2 weeks, how often have you been bothered by the following problems: thoughts that you would be better off dead or of hurting yourself in some way?"_

    **The variables `DPQ010` - `DPQ090` may have the same possible values:**
    - '0' - Not at all
    - '1' - Several days
    - '2' - More than half the days	
    - '3' - Nearly every day
    - '7' - Refused (to answer)
    - '9' - Don't know

11. DPQ100 - Difficulty these problems have caused.
    
    _The question is "How difficult have these problems made it for you to do your work, take care of things at home, or get along with people?"_

    **The variable has the possible values:**
    - '0' - Not at all difficult
    - '1' - Somewhat difficult
    - '2' - Very difficult
    - '3' - Extremely difficult
    - '7' - Refused (to answer)
    - '9' - Don't know

### Preparing the DPQ dataset

To calculate PHQ-9 total score, which is required to estimate depression levels, we need to select only valid answers to the first 9 symptom questions (`DPQ010` - `DPQ090`). Their values range from 0 to 3. However, before filtering participants, we need to clean and check the imported XPT values.


#### 1. Inspecting response values 

In [10]:
dpq.sample(10)

,SEQN,DPQ010,DPQ020,DPQ030,DPQ040,DPQ050,DPQ060,DPQ070,DPQ080,DPQ090,DPQ100
5436,140620.0,5.397605e-79,1.000000e+00,5.397605e-79,1.000000e+00,1.000000e+00,1.000000e+00,1.000000e+00,5.397605e-79,5.397605e-79,5.397605e-79
21,130406.0,5.397605e-79,5.397605e-79,5.397605e-79,5.397605e-79,1.000000e+00,5.397605e-79,1.000000e+00,5.397605e-79,5.397605e-79,5.397605e-79
4312,138469.0,5.397605e-79,5.397605e-79,1.000000e+00,1.000000e+00,5.397605e-79,5.397605e-79,5.397605e-79,5.397605e-79,5.397605e-79,5.397605e-79
3187,136425.0,5.397605e-79,1.000000e+00,5.397605e-79,1.000000e+00,5.397605e-79,1.000000e+00,5.397605e-79,5.397605e-79,5.397605e-79,5.397605e-79
4120,138119.0,5.397605e-79,5.397605e-79,5.397605e-79,5.397605e-79,5.397605e-79,5.397605e-79,5.397605e-79,5.397605e-79,5.397605e-79,NaN
5210,140185.0,2.000000e+00,1.000000e+00,1.000000e+00,2.000000e+00,1.000000e+00,2.000000e+00,5.397605e-79,5.397605e-79,5.397605e-79,1.000000e+00
2761,135622.0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
2688,135498.0,5.397605e-79,5.397605e-79,5.397605e-79,5.397605e-79,5.397605e-79,5.397605e-79,5.397605e-79,5.397605e-79,5.397605e-79,NaN
4529,138891.0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
3986,137879.0,5.397605e-79,5.397605e-79,1.000000e+00,1.000000e+00,5.397605e-79,5.397605e-79,5.397605e-79,5.397605e-79,5.397605e-79,5.397605e-79


In [118]:
dpq_columns = ['DPQ010', 'DPQ020', 'DPQ030', 'DPQ040','DPQ050','DPQ060','DPQ070', 'DPQ080', 'DPQ090', 'DPQ100']
for column in dpq_columns:
    print(dpq[column].value_counts(dropna=False).sort_index(), end='\n\n')
  

DPQ010
5.397605e-79    3703
1.000000e+00    1205
2.000000e+00     334
3.000000e+00     256
7.000000e+00       5
9.000000e+00      16
NaN              818
Name: count, dtype: int64

DPQ020
5.397605e-79    3674
1.000000e+00    1325
2.000000e+00     295
3.000000e+00     214
7.000000e+00       3
9.000000e+00       7
NaN              819
Name: count, dtype: int64

DPQ030
5.397605e-79    2772
1.000000e+00    1690
2.000000e+00     511
3.000000e+00     536
7.000000e+00       3
9.000000e+00       4
NaN              821
Name: count, dtype: int64

DPQ040
5.397605e-79    2348
1.000000e+00    2113
2.000000e+00     571
3.000000e+00     474
7.000000e+00       3
9.000000e+00       5
NaN              823
Name: count, dtype: int64

DPQ050
5.397605e-79    3710
1.000000e+00    1138
2.000000e+00     371
3.000000e+00     291
9.000000e+00       3
NaN              824
Name: count, dtype: int64

DPQ060
5.397605e-79    3971
1.000000e+00    1054
2.000000e+00     268
3.000000e+00     212
7.000000e+00       3
9.00

During the inspection, we observe a discrepancy between the NHANES documentation and the data. According to the NHANES codebook, participants could respond with '0', which means 'not at all'/'not at all difficult'. 

Nevertheless, we observe a '5' answer, which is not documented. 
The reason for that discrepancy might happen during importing the XPT files into the pandas library. 

The responses that should represent '0' were imported from the XPT file as extremely small positive floating-point values. These values do not correspond to any valid PHQ-9 response category and are treated as zeros.

#### 2. Cleaning imported 'zero' values

In [119]:
dpq_clean = dpq.copy()

dpq_clean[dpq_columns] = dpq_clean[dpq_columns].mask(
    dpq_clean[dpq_columns] < 1e-70,
    0
)

In [120]:
#Check the result

for column in dpq_columns:
    print(dpq_clean[column].value_counts(dropna=False).sort_index(), end='\n\n')
  

DPQ010
0.0    3703
1.0    1205
2.0     334
3.0     256
7.0       5
9.0      16
NaN     818
Name: count, dtype: int64

DPQ020
0.0    3674
1.0    1325
2.0     295
3.0     214
7.0       3
9.0       7
NaN     819
Name: count, dtype: int64

DPQ030
0.0    2772
1.0    1690
2.0     511
3.0     536
7.0       3
9.0       4
NaN     821
Name: count, dtype: int64

DPQ040
0.0    2348
1.0    2113
2.0     571
3.0     474
7.0       3
9.0       5
NaN     823
Name: count, dtype: int64

DPQ050
0.0    3710
1.0    1138
2.0     371
3.0     291
9.0       3
NaN     824
Name: count, dtype: int64

DPQ060
0.0    3971
1.0    1054
2.0     268
3.0     212
7.0       3
9.0       2
NaN     827
Name: count, dtype: int64

DPQ070
0.0    4036
1.0     989
2.0     277
3.0     203
7.0       2
9.0       1
NaN     829
Name: count, dtype: int64

DPQ080
0.0    4748
1.0     511
2.0     129
3.0     108
7.0       4
9.0       8
NaN     829
Name: count, dtype: int64

DPQ090
0.0    5207
1.0     209
2.0      57
3.0      28
9.0       5
N

#### 3. Selecting records with valid answers to symptom questions

In [121]:
phq9_columns = ['DPQ010', 'DPQ020', 'DPQ030', 'DPQ040','DPQ050','DPQ060','DPQ070', 'DPQ080', 'DPQ090']

valid_phq9 = dpq_clean[phq9_columns].isin([0, 1, 2, 3]).all(axis=1)

filtered_dpq = dpq_clean[valid_phq9].copy()

filtered_dpq


,SEQN,DPQ010,DPQ020,DPQ030,DPQ040,DPQ050,DPQ060,DPQ070,DPQ080,DPQ090,DPQ100
1,130379.0,0.0,0.0,1.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0
2,130380.0,0.0,0.0,1.0,1.0,0.0,0.0,0.0,0.0,0.0,0.0
3,130386.0,0.0,0.0,0.0,0.0,0.0,1.0,0.0,0.0,0.0,0.0
4,130387.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,NaN
6,130389.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,NaN
...,...,...,...,...,...,...,...,...,...,...,...
6331,142303.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,NaN
6333,142307.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,NaN
6334,142308.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,NaN
6335,142309.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,NaN


In [122]:
for column in dpq_columns:
    print(filtered_dpq[column].value_counts(dropna=False).sort_index(), end='\n\n')

DPQ010
0.0    3679
1.0    1195
2.0     331
3.0     250
Name: count, dtype: int64

DPQ020
0.0    3644
1.0    1313
2.0     289
3.0     209
Name: count, dtype: int64

DPQ030
0.0    2745
1.0    1674
2.0     508
3.0     528
Name: count, dtype: int64

DPQ040
0.0    2322
1.0    2100
2.0     566
3.0     467
Name: count, dtype: int64

DPQ050
0.0    3677
1.0    1123
2.0     369
3.0     286
Name: count, dtype: int64

DPQ060
0.0    3938
1.0    1040
2.0     267
3.0     210
Name: count, dtype: int64

DPQ070
0.0    4005
1.0     981
2.0     272
3.0     197
Name: count, dtype: int64

DPQ080
0.0    4717
1.0     507
2.0     126
3.0     105
Name: count, dtype: int64

DPQ090
0.0    5163
1.0     209
2.0      56
3.0      27
Name: count, dtype: int64

DPQ100
0.0    2547
1.0    1298
2.0     205
3.0      73
7.0       2
9.0       2
NaN    1328
Name: count, dtype: int64



### DPQ data preparation summary
The original DPQ dataset contained 6337 participants. We corrected the imported zero values and then excluded participants with incomplete responses to the nine PHQ-9 symptom questions. The final analytical sample consisted of 5455 participants (86.08% of the original sample).

In [123]:
print(f'Original sample: {len(dpq)}')
print(f'Analytical sample: {len(filtered_dpq)}')
print(f'Excluded: {len(dpq) - len(filtered_dpq)}')
print(f'Percentage of original sample: {len(filtered_dpq)/len(dpq):.2%}')

Original sample: 6337
Analytical sample: 5455
Excluded: 882
Percentage of original sample: 86.08%


### Merging DEMO and DPQ datasets

The analytical dataset was created by merging the filtered DEMO and DPQ datasets using the unique participant identifier - `SEQN`. The final dataset includes participants aged 20 or older who provided complete responses to all nine PHQ-9 symptom questions. It contains 5210 participants with no duplicate identifiers.

In [124]:
analysis_df = pd.merge(filtered_dpq, demo_adults, how = 'inner', on='SEQN')
analysis_df.info()

<class 'pandas.DataFrame'>
RangeIndex: 5210 entries, 0 to 5209
Data columns (total 24 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   SEQN      5210 non-null   float64
 1   DPQ010    5210 non-null   float64
 2   DPQ020    5210 non-null   float64
 3   DPQ030    5210 non-null   float64
 4   DPQ040    5210 non-null   float64
 5   DPQ050    5210 non-null   float64
 6   DPQ060    5210 non-null   float64
 7   DPQ070    5210 non-null   float64
 8   DPQ080    5210 non-null   float64
 9   DPQ090    5210 non-null   float64
 10  DPQ100    3918 non-null   float64
 11  RIDSTATR  5210 non-null   float64
 12  RIAGENDR  5210 non-null   float64
 13  RIDAGEYR  5210 non-null   float64
 14  RIDRETH3  5210 non-null   float64
 15  RIDEXMON  5210 non-null   float64
 16  DMQMILIZ  5210 non-null   float64
 17  DMDBORN4  5210 non-null   float64
 18  DMDYRUSR  937 non-null    float64
 19  DMDEDUC2  5210 non-null   float64
 20  DMDMARTZ  5210 non-null   float64
 21  RI

In [125]:
analysis_df['SEQN'].duplicated().sum()

np.int64(0)

### Missing data overview

#### 1. Misssing data summary

In [126]:
missing_summary = pd.DataFrame({
    'Available': analysis_df.notna().sum(),
    'Missing': analysis_df.isnull().sum()
})

missing_summary['Missing (%)'] = (
    missing_summary['Missing'] / len(analysis_df) * 100
).round(1)

missing_summary = missing_summary[missing_summary['Missing']>0]
missing_summary


,Available,Missing,Missing (%)
DPQ100,3918,1292,24.8
DMDYRUSR,937,4273,82.0
RIDEXPRG,940,4270,82.0
INDFMPIR,4586,624,12.0


#### 2. Investigating missing values

##### DPQ100 
`DPQ100` measures how much depressive symptoms interfere with daily life. Unlike the first nine questions, it is not included in the PHQ-9 total score.

In [127]:
dpq10_missing = analysis_df[analysis_df['DPQ100'].isnull()]
dpq10_missing

,SEQN,DPQ010,DPQ020,DPQ030,DPQ040,DPQ050,DPQ060,DPQ070,DPQ080,DPQ090,...,RIDRETH3,RIDEXMON,DMQMILIZ,DMDBORN4,DMDYRUSR,DMDEDUC2,DMDMARTZ,RIDEXPRG,DMDHHSIZ,INDFMPIR
3,130387.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,...,3.0,2.0,2.0,1.0,NaN,5.0,3.0,NaN,1.0,1.32
4,130389.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,...,3.0,1.0,2.0,2.0,6.0,5.0,1.0,NaN,2.0,5.00
8,130394.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,...,3.0,1.0,2.0,1.0,NaN,5.0,1.0,NaN,4.0,5.00
12,130398.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,...,1.0,2.0,2.0,2.0,6.0,3.0,1.0,NaN,2.0,1.67
17,130407.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,...,3.0,1.0,2.0,1.0,NaN,5.0,2.0,NaN,2.0,4.37
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
5204,142302.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,...,3.0,2.0,2.0,1.0,NaN,4.0,3.0,NaN,2.0,2.33
5205,142303.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,...,7.0,2.0,1.0,1.0,NaN,3.0,2.0,NaN,2.0,0.98
5206,142307.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,...,4.0,2.0,2.0,1.0,NaN,5.0,3.0,NaN,5.0,NaN
5207,142308.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,...,2.0,1.0,2.0,2.0,6.0,4.0,1.0,NaN,3.0,1.95


In [128]:
phq9_columns = ['DPQ010', 'DPQ020', 'DPQ030', 'DPQ040','DPQ050','DPQ060','DPQ070', 'DPQ080', 'DPQ090']
phq9_all_zero = dpq10_missing[
    (dpq10_missing[phq9_columns] == 0).all(axis=1)
]
phq9_all_zero

,SEQN,DPQ010,DPQ020,DPQ030,DPQ040,DPQ050,DPQ060,DPQ070,DPQ080,DPQ090,...,RIDRETH3,RIDEXMON,DMQMILIZ,DMDBORN4,DMDYRUSR,DMDEDUC2,DMDMARTZ,RIDEXPRG,DMDHHSIZ,INDFMPIR
3,130387.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,...,3.0,2.0,2.0,1.0,NaN,5.0,3.0,NaN,1.0,1.32
4,130389.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,...,3.0,1.0,2.0,2.0,6.0,5.0,1.0,NaN,2.0,5.00
8,130394.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,...,3.0,1.0,2.0,1.0,NaN,5.0,1.0,NaN,4.0,5.00
12,130398.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,...,1.0,2.0,2.0,2.0,6.0,3.0,1.0,NaN,2.0,1.67
17,130407.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,...,3.0,1.0,2.0,1.0,NaN,5.0,2.0,NaN,2.0,4.37
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
5204,142302.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,...,3.0,2.0,2.0,1.0,NaN,4.0,3.0,NaN,2.0,2.33
5205,142303.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,...,7.0,2.0,1.0,1.0,NaN,3.0,2.0,NaN,2.0,0.98
5206,142307.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,...,4.0,2.0,2.0,1.0,NaN,5.0,3.0,NaN,5.0,NaN
5207,142308.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,...,2.0,1.0,2.0,2.0,6.0,4.0,1.0,NaN,3.0,1.95


In [129]:
len(phq9_all_zero) == len(dpq10_missing)

False

In [130]:
not_all_zero = dpq10_missing[
    ~(dpq10_missing[phq9_columns] == 0).all(axis=1)
]

not_all_zero[phq9_columns + ['DPQ100'] + ['SEQN']]

,DPQ010,DPQ020,DPQ030,DPQ040,DPQ050,DPQ060,DPQ070,DPQ080,DPQ090,DPQ100,SEQN
4007,0.0,0.0,0.0,2.0,3.0,0.0,0.0,0.0,0.0,NaN,139522.0


In [131]:
all_symptoms_zero = (
    dpq10_missing[phq9_columns]
    .eq(0)
    .all(axis=1)
)

print(f'Missing DPQ100: {len(dpq10_missing)}')
print(f'Structural missing values: {all_symptoms_zero.sum()}')
print(f'Other missing values: {(~all_symptoms_zero).sum()}')

Missing DPQ100: 1292
Structural missing values: 1291
Other missing values: 1


##### DPQ100 summary
Of the 1292 participants with missing `DPQ100` values, **1291** reported 'Not at all' to all nine PHQ-9 symptom questions. Since `DPQ100` is asked only when at least one symptom is endorsed (according NHANES documentation), these _1291 missing values are structural_. One participant reported depressive symptoms but still had a missing `DPQ100` value, which appears to be a non-structural missing response. 

Since the participant (`SEQN = 139522.0`) has completed PHQ-9 responses, they are retained in the analytical dataset. However, this exaptional case will be taken into account in future analyses involving 'DPQ100'.

##### DMDYRUSR
`DMDYRUSR` reflects the length of time the participant has lived in the US. According to NHANES documentation, only participants born outside the US were asked the month and year when they came to live in the US. 
Let's verify that the missing values are structural.

In [132]:
dmdyrusr_missing = analysis_df[analysis_df['DMDYRUSR'].isnull()]
dmdyrusr_missing

,SEQN,DPQ010,DPQ020,DPQ030,DPQ040,DPQ050,DPQ060,DPQ070,DPQ080,DPQ090,...,RIDRETH3,RIDEXMON,DMQMILIZ,DMDBORN4,DMDYRUSR,DMDEDUC2,DMDMARTZ,RIDEXPRG,DMDHHSIZ,INDFMPIR
0,130379.0,0.0,0.0,1.0,0.0,0.0,0.0,0.0,0.0,0.0,...,3.0,2.0,2.0,1.0,NaN,5.0,1.0,NaN,2.0,5.00
2,130386.0,0.0,0.0,0.0,0.0,0.0,1.0,0.0,0.0,0.0,...,1.0,1.0,2.0,1.0,NaN,4.0,1.0,NaN,3.0,1.33
3,130387.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,...,3.0,2.0,2.0,1.0,NaN,5.0,3.0,NaN,1.0,1.32
5,130391.0,3.0,3.0,3.0,3.0,3.0,3.0,3.0,2.0,1.0,...,3.0,2.0,2.0,1.0,NaN,3.0,3.0,2.0,3.0,0.93
6,130392.0,0.0,0.0,0.0,1.0,0.0,0.0,0.0,0.0,0.0,...,3.0,2.0,2.0,1.0,NaN,5.0,1.0,NaN,2.0,3.04
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
5204,142302.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,...,3.0,2.0,2.0,1.0,NaN,4.0,3.0,NaN,2.0,2.33
5205,142303.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,...,7.0,2.0,1.0,1.0,NaN,3.0,2.0,NaN,2.0,0.98
5206,142307.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,...,4.0,2.0,2.0,1.0,NaN,5.0,3.0,NaN,5.0,NaN
5208,142309.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,...,2.0,1.0,1.0,1.0,NaN,4.0,2.0,NaN,5.0,3.11


Since `DMDYRUSR` was collected only for participants who were born abroad, we can examine `DMDBORN4` variable, that indicates the participant's place of birth. `DMDBORN4` has two categories:
1. Born in 50 U.S. states or Washington, DC.
2. Born in other countries, including U.S. territories.

As expected, `DMDYRUSR` is missing for participants born in the US. Let's verify whether any participants classified as `DMDBORN4 = 2` (born in other countries or U.S. territories) also have missing `DMDYRUSR` values.

In [133]:
dmdborn4_category2 = dmdyrusr_missing[dmdyrusr_missing['DMDBORN4']!=1]
dmdborn4_category2

,SEQN,DPQ010,DPQ020,DPQ030,DPQ040,DPQ050,DPQ060,DPQ070,DPQ080,DPQ090,...,RIDRETH3,RIDEXMON,DMQMILIZ,DMDBORN4,DMDYRUSR,DMDEDUC2,DMDMARTZ,RIDEXPRG,DMDHHSIZ,INDFMPIR


##### DMDYRUSR summary
No participants with missing `DMDYRUSR` belonged to `DMDBORN4` category 2. Therefore, all missing `DMDYRUSR` values correspond to participants in `DMDBORN4` category 1, indicating that the missing values are structural rather than missing responses.

##### RIDEXPRG
`RIDEXPRG` represents pregnancy status. This data was released only for women aged 20-44 years. If a woman didn't know she was pregnant or didn't participate in the examination, the pregnancy status was assigned a value of '3', which means 'could not be determined'.
Thus, the structural missing data must be for men and women above 44 years old in our analysis dataset. 
Let's verify that the missing values are structural.

In [134]:
ridexprg_missing = analysis_df[analysis_df['RIDEXPRG'].isnull()]
ridexprg_missing[['SEQN','RIAGENDR','RIDAGEYR','RIDEXPRG']]

,SEQN,RIAGENDR,RIDAGEYR,RIDEXPRG
0,130379.0,1.0,66.0,NaN
2,130386.0,1.0,34.0,NaN
3,130387.0,2.0,68.0,NaN
4,130389.0,1.0,59.0,NaN
6,130392.0,2.0,74.0,NaN
...,...,...,...,...
5205,142303.0,2.0,69.0,NaN
5206,142307.0,2.0,49.0,NaN
5207,142308.0,1.0,50.0,NaN
5208,142309.0,1.0,40.0,NaN


In [135]:
men_and_older_women = ridexprg_missing[(ridexprg_missing['RIAGENDR']==1) | (ridexprg_missing['RIDAGEYR']>44)]
men_and_older_women


,SEQN,DPQ010,DPQ020,DPQ030,DPQ040,DPQ050,DPQ060,DPQ070,DPQ080,DPQ090,...,RIDRETH3,RIDEXMON,DMQMILIZ,DMDBORN4,DMDYRUSR,DMDEDUC2,DMDMARTZ,RIDEXPRG,DMDHHSIZ,INDFMPIR
0,130379.0,0.0,0.0,1.0,0.0,0.0,0.0,0.0,0.0,0.0,...,3.0,2.0,2.0,1.0,NaN,5.0,1.0,NaN,2.0,5.00
2,130386.0,0.0,0.0,0.0,0.0,0.0,1.0,0.0,0.0,0.0,...,1.0,1.0,2.0,1.0,NaN,4.0,1.0,NaN,3.0,1.33
3,130387.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,...,3.0,2.0,2.0,1.0,NaN,5.0,3.0,NaN,1.0,1.32
4,130389.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,...,3.0,1.0,2.0,2.0,6.0,5.0,1.0,NaN,2.0,5.00
6,130392.0,0.0,0.0,0.0,1.0,0.0,0.0,0.0,0.0,0.0,...,3.0,2.0,2.0,1.0,NaN,5.0,1.0,NaN,2.0,3.04
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
5205,142303.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,...,7.0,2.0,1.0,1.0,NaN,3.0,2.0,NaN,2.0,0.98
5206,142307.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,...,4.0,2.0,2.0,1.0,NaN,5.0,3.0,NaN,5.0,NaN
5207,142308.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,...,2.0,1.0,2.0,2.0,6.0,4.0,1.0,NaN,3.0,1.95
5208,142309.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,...,2.0,1.0,1.0,1.0,NaN,4.0,2.0,NaN,5.0,3.11


In [136]:
len(ridexprg_missing) == len(men_and_older_women)

True

##### RIDEXPRG summary
All missing values in `RIDEXPRG` are structural. There are no women aged 20-44 with missing `RIDEXPRG` values.


##### INDFMPIR
`INDFMPIR` represents the ratio of family income to poverty level. According to the NHANES documentation, `INDFMPIR` was not computed when detailed income information was unavailable or when income data were missing. Therefore, unlike the previous variables, the missing values in `INDFMPIR` are not fully explained by the questionnaire design and should be considered genuine missing values. Let's briefly look at missing data.

In [137]:
indfmpir_missing = analysis_df[analysis_df['INDFMPIR'].isnull()]
indfmpir_missing

,SEQN,DPQ010,DPQ020,DPQ030,DPQ040,DPQ050,DPQ060,DPQ070,DPQ080,DPQ090,...,RIDRETH3,RIDEXMON,DMQMILIZ,DMDBORN4,DMDYRUSR,DMDEDUC2,DMDMARTZ,RIDEXPRG,DMDHHSIZ,INDFMPIR
22,130416.0,0.0,0.0,1.0,0.0,0.0,0.0,0.0,0.0,0.0,...,2.0,1.0,2.0,2.0,6.0,3.0,2.0,NaN,2.0,NaN
46,130478.0,0.0,0.0,1.0,2.0,1.0,0.0,0.0,0.0,0.0,...,3.0,2.0,2.0,1.0,NaN,5.0,1.0,NaN,2.0,NaN
50,130493.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,...,3.0,2.0,2.0,1.0,NaN,5.0,3.0,NaN,2.0,NaN
64,130525.0,0.0,0.0,1.0,0.0,0.0,0.0,0.0,0.0,0.0,...,3.0,1.0,2.0,2.0,6.0,5.0,1.0,NaN,3.0,NaN
71,130541.0,1.0,1.0,0.0,2.0,0.0,1.0,0.0,0.0,0.0,...,3.0,1.0,2.0,2.0,2.0,5.0,1.0,2.0,3.0,NaN
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
5167,142226.0,1.0,0.0,0.0,1.0,0.0,0.0,0.0,0.0,0.0,...,4.0,2.0,2.0,2.0,3.0,2.0,2.0,NaN,2.0,NaN
5178,142247.0,1.0,1.0,0.0,1.0,0.0,0.0,0.0,0.0,0.0,...,3.0,2.0,2.0,2.0,4.0,5.0,1.0,NaN,3.0,NaN
5184,142263.0,0.0,0.0,0.0,1.0,0.0,0.0,0.0,0.0,0.0,...,3.0,2.0,2.0,1.0,NaN,5.0,2.0,2.0,1.0,NaN
5197,142289.0,0.0,0.0,1.0,0.0,0.0,0.0,0.0,0.0,0.0,...,3.0,2.0,2.0,1.0,NaN,5.0,1.0,NaN,2.0,NaN


##### INDFMPIR summary
Participants with missing `INDFMPIR` values are retained in the analytical dataset because the current step focuses on data exploration rather than statistical analysis. The missing values will be addressed when analyzing the association between income and depressive symptoms.

#### 3. Missing Data Conclusions
Investigating missing values revealed two missing data mechanisms: structural missingness and genuine missingness.

`DPQ100`: 1291 structural missing values and 1 genuine missing value.
`DMDYRUSR`: 4273 structural missing values.
`RIDEXPRG`: 4270 structural missing values.
`INDFMPIR`: 624 genuine missing values.

In [138]:
missing_mechanisms = pd.DataFrame({
    'Variable':['DPQ100', 'DMDYRUSR', 'RIDEXPRG', 'INDFMPIR'],
    'Missing mechanism':['Mostly structural (1 exception)', 'Structural','Structural', 'Genuine'],
},index = [1,2,3,4])

missing_mechanisms

,Variable,Missing mechanism
1,DPQ100,Mostly structural (1 exception)
2,DMDYRUSR,Structural
3,RIDEXPRG,Structural
4,INDFMPIR,Genuine
